deepmd/entrypoints/test.py

```diff
@@ -307,6 +384,11 @@ def test_ener(
         data.add("hessian", 1, atomic=True, must=True, high_prec=False)
 
     test_data = data.get_test()
+    find_energy = test_data.get("find_energy")
+    find_force = test_data.get("find_force")
+    find_virial = test_data.get("find_virial")
+    find_force_r = test_data.get("find_force_real")
+    find_force_m = test_data.get("find_force_mag")
```

Comment:
Have you checked the correct keys for `force_r` and `force_m` in spin model ?
See `deepmd/pt/loss/ener_spin.py` for details. 
They should be `find_force` and `find_force_mag`.